### PR TITLE
Release a new version for each master merge

### PR DESCRIPTION
**Pull request by evalsocket**
# TL;DR
- Updated GH workflow for releasing flytectl new version for each master merge

Tested Result: https://github.com/evalsocket/flytectl/actions/runs/941791206
## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [x] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/lyft/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/lyft/flyte/issues/<number>_

### CODE REVIEW

File: .github/workflows/master.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  bump-version:
+    name: Bump Version
+    if: github.event.commits[0].author.name != 'flyte-bot'
```
Comment (pmahindrakar-oss) [inline .github/workflows/master.yml:11]: Can we use head_commit from the event instead.
https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#push

**Review by pmahindrakar-oss — DISMISSED** [review]

File: .github/workflows/master.yml (hunk 1 above)
Comment (evalsocket) [reply .github/workflows/master.yml]: done

**Review by pmahindrakar-oss — APPROVED** [review]
